modules/@apostrophecms/login/index.js

```diff
@@ -981,6 +996,87 @@ module.exports = {
             required: true
           })
           );
+      },
+
+      normalizeLoginName(usernameOrEmail) {
+        if (typeof usernameOrEmail !== 'string' || !self.options.caseInsensitive) {
+          return usernameOrEmail;
+        }
+        return usernameOrEmail.toLowerCase();
+      },
+
+      async manageLoginCaseInsensitiveMigration() {
+        if (self.options.caseInsensitive) {
+          self.apos.migration.add('login-case-insensitive', async () => {
+            await self.caseInsensitiveTask();
+          });
+        } else {
+          self.apos.migration.cancel('login-case-insensitive');
+        }
+      },
+
+      async caseInsensitiveTask() {
+        const duplicatedUsernames = [];
+        await self.apos.migration.eachDoc({ type: '@apostrophecms/user' }, 1, async (user) => {
+          const usernameLower = user.username.toLowerCase();
+          const emailLower = user.email?.toLowerCase();
+          const shouldUpdateUsername = user.username !== usernameLower;
+          const shouldUpdateEmail = user.email && user.email !== emailLower;
+
+          if (!shouldUpdateUsername && !shouldUpdateEmail) {
+            return;
+          }
+          const existingUsername = await self.apos.doc.db.findOne({
```

Comment:
Not just a username issue. The username, the email, or both could lead to duplication and therefore trigger a unique key error.